modules/common/src/main/scala/surge/internal/health/windows/stream/sliding/SlidingHealthSignalStream.scala

```diff
@@ -78,28 +79,25 @@ private class SlidingHealthSignalStreamImpl(
   }
 
   override def processWindows(signalPatternMatchers: Seq[SignalPatternMatcher], monitoringActor: Option[ActorRef]): StreamHandle = {
+    // Create a windowActor for each configured windowing frequency
     val windowActors: Seq[HealthSignalWindowActorRef] = {
       val advancerConfig = windowingConfig.advancerConfig
       windowingConfig.frequencies
         .map(freq => HealthSignalWindowActor(actorSystem, freq, WindowSlider(advancerConfig.advanceAmount, advancerConfig.buffer)))
         .map(ref => ref.start(monitoringActor))
     }
 
-    // todo: consider using an akka source
-    val future = Executors
-      .newCachedThreadPool()
-      .submit(new Callable[Unit] {
-        override def call: Unit = {
-          while (true) {
-            val signal = signals.take()
-            windowActors.foreach(a => a.processSignal(signal))
-          }
-        }
-      })
+    val blockingSource = Source.unfoldResource[HealthSignal, ArrayBlockingQueue[HealthSignal]](() => signals, signals => Option(signals.take()), _ => {})
```

Comment:
This seems like a bit of an awkward fit here since the `unfoldResource` is really meant for wrapping bounded things, like querying a bunch of records from a DB. I think maybe a `Source.queue` would be a better fit - we could bound it to something decently large so we likely won't drop elements but wouldn't grow unbounded. It may require a bit of plumbing to the superclass that feeds signals in. I think that should also eliminate the need for a RestartSource here.

Reply:
looking into it.

Reply:
done